### ISSUE TITLE
arbitrary_source_item_ordering reports warning: unknown lint with 1.83.0 (docs say this lint was enabled in 1.82.0)

### ISSUE THREAD

**Issue by ShakenCodes** [rust-lang/rust-clippy#13818]
### Summary

While using the 1.83.0 rust toolchain, adding the line `#![deny(clippy::arbitrary_source_item_ordering)]` to the lib.rs file, generates the following unexpected warning:
```bash
% cargo clippy
warning: unknown lint: `clippy::arbitrary_source_item_ordering`
 --> src/lib.rs:1:9
  |
1 | #![deny(clippy::arbitrary_source_item_ordering)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unknown_lints)]` on by default

warning: `replicate_arbitrary_source_item_ordering` (lib) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.03s
%
```

### Reproducer

I tried this code:

```rust
#![deny(clippy::arbitrary_source_item_ordering)]

pub fn add(left: u64, right: u64) -> u64 {
    left + right
}
```

I expected to see this happen:

Combpilation with no errors or warning

Instead, this happened:

A warning was emitted claiming this is an unknown lint. The Clippy documentation claims this was introduced in the previous release (1.82.0), while these examples are compiled with the 1.83.0 toolchain on Mac.

```bash
% cargo clippy
warning: unknown lint: `clippy::arbitrary_source_item_ordering`
 --> src/lib.rs:1:9
  |
1 | #![deny(clippy::arbitrary_source_item_ordering)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unknown_lints)]` on by default

warning: `replicate_arbitrary_source_item_ordering` (lib) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.03s
%
```

### Version

```text
rustc 1.83.0 (90b35a623 2024-11-26)
binary: rustc
commit-hash: 90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf
commit-date: 2024-11-26
host: aarch64-apple-darwin
release: 1.83.0
LLVM version: 19.1.1
```

**Comment by lapla-cogito** [comment]
This is probably a documentation issue. Each lint uses an attribute to indicate which version it is included from, and I believe this is due to the fact that this part of the documentation is different from the actual documentation. In fact, this lint exists in the following environment (beta channel):
```
$ cargo clippy --version
clippy 0.1.84 (202008a1b8 2024-12-07)

$ cargo clippy --explain arbitrary_source_item_ordering
### What it does

Confirms that items are sorted in source files as per configuration.

### Why restrict this?

Keeping a consistent ordering throughout the codebase helps with working
as a team, and possibly improves maintainability of the codebase. The
...
```
